ocaml/xapi/network_event_loop.ml

```diff
@@ -0,0 +1,143 @@
+module D=Debug.Make(struct let name="network_event_loop" end)
+
+module StringMap = Map.Make(struct type t = string let compare = compare end)
+
+type 'a update = Deleted | Snapshot of 'a
+
+let _watch_networks __context ~update_firewall ~wait_after_failure_seconds =
+
+  (* We keep track of the network objects in the database using this event loop. *)
+  let classes = ["network"] in
+  (* We store the networks in a reference to record map. *)
+  let networks = StringMap.empty in
+  (* We keep track of the interfaces that we last passed to the firewall script
+     to allow NBD traffic on them. At startup, NBD is blocked on all
+     interfaces. *)
+  let allowed_interfaces = [] in
+
+  let api_timeout = 60. in
+  let timeout = 30. +. api_timeout +. !Db_globs.master_connection_reset_timeout in
+
+  let rec update_networks ~token ~networks =
+
+    let most_recent_state events =
+      let events_from_newest_to_oldest =
```

Comment:
Are you sure we can't rely on them being sorted already?

Reply:
This intermediate commit has been changed by a subsequent one - I though I'd just keep it in case we decide to use caching - sorry for the confusion.

Reply:
However, to answer your question, it's not mentioned in the XenAPI reference, and I chose not to rely on assumptions that are not part of our official interface/contract/documentation.

Reply:
There's no guarantee of ordering of events, but you'll get each object only once, containing the final state of the object.